### ISSUE TITLE
 Feature - execute plugin on empty block

### ISSUE THREAD

**Issue by barak007** [thysultan/stylis.js#41]
I need a way to enable plugins to execute on empty blocks.

I see two use cases for this: 

1. transform and preserve any selector that was in my source (lossless).
2. when used as linter be able to warn on empty selector.

The issue: 
```css
.selector {}
```
This will not trigger my plugin.

---

I did fixed it in my project by modifying the source.

I added config called lossless 
```js
stylis.use({lossless: true})
```

and changed this code block

```js
// execute plugins, block context
if ((lossless && current[0] !== '') || (length > 0 && plugged > 0)) {
    result = proxy(BLCKS, out, current, parent, line, column, length, id)

    if (result !== void 0) {
        length = (out = result).length
    }
}
```

I can submit a pull request but I wanted to create an issue first.

10x in advance.

**Comment by thysultan** [comment]
Yes this would be useful for a linter plugins.

So either this could be a general setting that you can set i.e `stylis.set({preserve: true})` that ensure it doesn't discard empty selectors or like you mentioned just confined to the execution of plugins.

Either way it would have to not confuse the `h1` in the nested block `h1 {h2 {color:red;}}` for an empty block.

**Comment by barak007** [comment]
Nice. so how do you like to do it? I like the "preserve" option more then lossless.

**Comment by thysultan** [comment]
I think the general approach`stylis.set({preserve: true})`. Will add this to the next minor release.

**Comment by thysultan** [comment]
This has landed in 3.2.0 using `stylis.set({preserve: true})`.

**Comment by barak007** [comment]
great!!! 10x
